Black-Market/Black-Market/Network/NetworkService.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  NetworkService.swift
+//  Black-Market
+//
+//  Created by Junior Sancho on 8/2/23.
+//
+
+import Foundation
+import Combine
+
+class NetworkService {
+    func getData(from endpoint: NetworkEndpoint) -> AnyPublisher<Data, Error> {
+        guard let request = buildRequest(from: endpoint) else {
+            let error = NetworkError.network(description: "Couldn't create URLRequest")
+            return Fail(error: error)
+                .eraseToAnyPublisher()
+        }
+        
+        return loadData(with: request)
+            .eraseToAnyPublisher()
+    }
+    
+    private func buildRequest(from endpoint: NetworkEndpoint) -> URLRequest? {
+        guard var urlComponents = URLComponents(string: endpoint.absoluteURL) else {
+            return nil
+        }
+        if endpoint.httpMethod == .get {
+            urlComponents.queryItems = endpoint.params.compactMap({ param -> URLQueryItem in
+                return URLQueryItem(name: param.key, value: param.value as? String ?? "")
+            })
+        }
+        guard let url = urlComponents.url else {
+            return nil
+        }
+        var urlRequest = URLRequest(url: url,
+                                    cachePolicy: .reloadIgnoringLocalCacheData,
+                                    timeoutInterval: 30)
```

Comment:
please use the indentation style recommended in our style guides:
```suggestion
        var urlRequest = URLRequest(
            url: url,
            cachePolicy: .reloadIgnoringLocalCacheData,
            timeoutInterval: 30
        )
```

Reply:
Done